### PR TITLE
Math Magicians: Tests

### PR DESCRIPTION
**Pull request by adarachel**
- [x] Set up React Testing Library.
- [x] Follow the instructions in the official [documentation](https://testing-library.com/docs/react-testing-library/intro).
- [x] You don't need to setup Jest (it's included in Create React App).
- [x] Write unit tests for the files operate.js and calculate.js using Jest.
- [x] Create unit tests for all React components:
- [x] Use Jest snapshots to test the components.
- [x] Use React Testing Library to simulate user interaction.
- [x] Run your full tests suite using Jest. All tests should pass.

### CODE REVIEW

File: src/tests/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import calculate from '../logic/calculate';
+
+test('calculate function adds two numbers', () => {
+  const result = calculate({ total: '2', next: '3', operation: '+' }, '=');
+  expect(result.total).toBe('5');
+});
+
+// Write more tests for the calculate function based on your logic
```
Comment (jerryowusu) [inline src/tests/calculate.test.js]: Nice work so far. However, as part of the project requirements, kindly make sure that all functionalities and scenarios of the calculate.js modules are covered using jest. :wink:

File: src/tests/operate.test.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import operate from '../logic/operate';
+
+test('operate function performs the correct operation', () => {
+  const result = operate('2', '3', '+');
+  expect(result).toBe('5');
+});
+
+// Write more tests for the operate function based on your logic
```
Comment (jerryowusu) [inline src/tests/operate.test.js]: Nice work so far. However, as part of the project requirements, kindly make sure that all functionalities and scenarios of the operate.js modules are covered using jest. :wink:

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @adarachel , 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

so-close

## Highlights
Some of the test are passing :+1: 
Pair programming activity :+1: 

## Required Changes ♻️

- [ ] Some of the test are failing as shown below. Kindly check and fix it to make sure all the test are passing before you submit for another review :+1: 

Screenshot from 2023-04-15 11-28-57


_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/Quote.test.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Quotes from '../components/Quote';
+
+it('matches snapshot', () => {
+  const tree = renderer.create(
+    <Quotes />,
+  ).toJSON();
+  expect(tree).toMatchSnapshot();
```
Comment (ghost) [inline src/tests/Quote.test.js:9]: - [ ] One of your test snapshots is failing. Kindly check it out

image

**Review by ghost — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!




## Highlights

✅ Correct use of Gitflow
✅ No linter error


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/tests/Home.test.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+import renderer from 'react-test-renderer';
+
+import Home from '../components/Home';
+
+it('matches home snapshot', () => {
+  const tree = renderer.create(
+    <Home />,
+  ).toJSON();
+  expect(tree).toMatchSnapshot();
+});
```
Comment (nahnahdev) [inline src/tests/Home.test.js:11]: - Great job so far @adarachel But currently, 3 of your tests are failing. So kindly ensure that all your tests are well written and pass successfully :100: 

image

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi  @adarachel , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which haven't been addressed yet.


 ## Highlights
- Most requested changes implemented :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @adarachel,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**